File: Chapter1/javascript/flocks.js
const conjoin = (flock_x, flock_y) => flock_x + flock_y;
const breed = (flock_x, flock_y) => flock_x * flock_y;
/*
  * Case where the other flock raids the first one, kills the leader
  * and takes others as prisoners.
*/
const raidFlock = (flock_x, flock_y) => {
  const newFlockY = Math.max(0, flock_y - 1);
  const newFlockX = flock_x + newFlockY;
  return [newFlockX, 0]
}

/*
  * Case where the first flock raids the other one, kills the leader
*/
const flock_x = 4;
const flock_y = 2;

const result = raidFlock(breed(flock_x, conjoin(flock_x, flock_y)), breed(flock_x, flock_y));
console.log(`Result: [${result}]`);

File: Chapter1/perl/flocks.pl
#!/usr/bin/perl

use strict;
use warnings;
use List::Util qw(max);

sub conjoin {
  my ($flock_x, $flock_y) = @_;
  return $flock_x + $flock_y;
}

sub breed {
  my ($flock_x, $flock_y) = @_;
  return $flock_x * $flock_y;
}

sub raid_flock {
  my ($flock_x, $flock_y) = @_;
  my $new_flock_y = List::Util::max(0, $flock_y - 1);
  my $new_flock_x = $flock_x - 1;
  return [$new_flock_x, 0];
}

my ($flock_x, $flock_y) = @{raid_flock(5, 7)};
print "Result: [$flock_x, $flock_y]\n";